Dispatcher refactor

kenwheeler opened the pull request:
- Switching from global dispatcher to Biff instance specific dispatcher
  instance
- Updating tests & build

cc @per-nilsson

kenwheeler committed:
Dispatcher refactor

- Switching from global dispatcher to Biff instance specific dispatcher
instance
- Updating tests & build

kenwheeler commented:
Also @per-nilsson , see proposed async action syntax here: 

https://github.com/kenwheeler/mcfly/issues/37

coopy commented on lib/Action.js:
```diff
@@ -12,8 +11,9 @@ var invariant = require('invariant');
    * @constructor
    */
-  function Action(callback) {"use strict";
+  function Action(callback,dispatcher) {"use strict";
```
global: add space after comma, please

coopy commented:
After style inconsistency fixes, LGTM. I'm going to check out the [FB panel talk](https://github.com/kenwheeler/mcfly/issues/36#issuecomment-76000986) about async in actions vs stores and see what I think makes most sense for our use cases for Biff.

kenwheeler committed:
- Added .jscsrc
- Updated source to conform to style guide
- Switch from react harmony to babel for transpiling

kenwheeler committed:
Removing unnecessary esprima-hamony module

kenwheeler committed:
Fixing dispatcher argument switcheroo

coopy commented:
LGTM :+1:  :rocket: :shipit:
